feat: statically analyse universal pages and layouts v3 (#13684)
sveltejs/kit: documentation/docs/20-core-concepts/40-page-options.md

If you add `export const ssr = false` to your root `+layout.js`, your entire app will only be rendered on the client — which essentially means you turn your app into an SPA.

> > [!NOTE] If you have a page option that isn't a boolean or string literal, even with `ssr` set to `false`, code that relies on browser APIs should be imported in your `+page.svelte` or `+layout.svelte` file instead. This is because complex values for page options need to be evaluated by importing your `+page.js` or `+layout.js` file on the server (if you have a runtime) and at build time.

Reviewer: ```suggestion
> [!NOTE] If all your page options are boolean or string literal values, SvelteKit will evaluate them statically. If not, it will import your `+page.js` or `+layout.js` file on the server (both at build time, and at runtime if your app isn't fully static) so it can evaluate the options. In the second case, browser-only code must not run when the module is loaded. In practice, this means you should import browser-only code in your `+page.svelte` or `+layout.svelte` file instead.
```

> > [!NOTE] If you have a page option that isn't a boolean or string literal, even with `ssr` set to `false`, code that relies on browser APIs should be imported in your `+page.svelte` or `+layout.svelte` file instead. This is because page options can be overridden and need to be evaluated by importing your `+page.js` or `+layout.js` file on the server (if you have a runtime) and at build time.

Author: I removed "(in case of prerendering)" because we still import those universal modules to analyse the page options regardless of prerendering.

> > [!NOTE] If you have a page option that isn't a boolean or string literal, even with `ssr` set to `false`, code that relies on browser APIs should be imported in your `+page.svelte` or `+layout.svelte` file instead. This is because page options can be overridden and need to be evaluated by importing your `+page.js` or `+layout.js` file on the server (if you have a runtime) and at build time.

Reviewer 2: ```suggestion
> [!NOTE] If you have a page option that isn't a boolean or string literal, even with `ssr` set to `false`, code that relies on browser APIs should be imported in your `+page.svelte` or `+layout.svelte` file instead. This is because complex values for page options need to be evaluated by importing your `+page.js` or `+layout.js` file on the server (if you have a runtime) and at build time.
```
